rust/src/shards/gui/containers/window.rs

```diff
@@ -245,6 +270,15 @@ impl Shard for Window {
     if !self.contents.is_empty() {
       let title: &str = self.title.get().try_into()?;
       let mut window = egui::Window::new(title);
+      if let Ok(id) = <&str>::try_from(self.id.get()) {
+        if let Some(id) = self.cached_id {
+          window = window.id(id);
+        } else {
+          let id = egui::Id::new(id);
+          window = window.id(id);
+          self.cached_id = Some(id);
+        }
```

Comment:
I think this can be written also like this:
```rust
let id = self.cached_id.get_or_insert_with(|| egui::Id::new(id));
window = window.id(id);
```

Reply:
I will have a look

Reply:
hmm sort of, but I prefer the clarity actually, functional rust is weird